build.gradle

```diff
@@ -73,3 +73,5 @@ subprojects {
 
 // Override spring boot's kotllin version dependency
 ext['kotlin.version'] = '${kotlinVersion}'
+
+assert JavaVersion.current().isJava11Compatible()
```

Comment:
Hmmm, surprising that we didn't already having something like this for Java 8.

Reply:
The build probably had failed with JDK 1.7 anyhow, so no need to check, I guess. Right now, we can build the project with Java 8 just fine AFAICT. This assertion will block using JDK/JRE 8 for any gradle tasks, so prevents running into any possible untested issues later at runtime, etc.